pkg/system/phase4_configuring.go

```diff
@@ -1425,6 +1429,76 @@ func (r *Reconciler) ReconcileOBCStorageClass() error {
 	return nil
 }
 
+// ReconcileODFBucketLoggingPVC ensures the bucket logging PVC is properly configured for 'guaranteed' logging
+func (r *Reconciler) ReconcileODFBucketLoggingPVC() error {
+	log := r.Logger.WithField("func", "ReconcileBucketLoggingPVC")
+
+	// Return if Bucket logging type is either not specified or not set to 'guaranteed'
+	if r.NooBaa.Spec.BucketLogging.LoggingType != nbv1.BucketLoggingTypeGuaranteed {
+		log.Info("Bucket logging type is either not specified or not set to 'Guaranteed'. skipping ReconcileODFBucketloggingPVC")
+		return nil
+	}
+
+	// Return if bucket logging PVC already exist and support RWX access mode
+	if r.NooBaa.Spec.BucketLogging.BucketLoggingPVC != nil {
+		r.BucketLoggingPVC.Name = *r.NooBaa.Spec.BucketLogging.BucketLoggingPVC
+		if util.KubeCheck(r.BucketLoggingPVC) {
+			for _, accessMode := range r.BucketLoggingPVC.Spec.AccessModes {
+				if accessMode == corev1.ReadWriteMany {
+					log.Info("Bucket logging PVC already exists and supports RWX mode. Skipping ReconcileODFBucketLoggingPVC.")
+					return nil
+				}
+			}
+			return util.NewPersistentError("InvalidBucketLoggingConfiguration",
+				"'Guaranteed' BucketLogging requires a Persistent Volume Claim (PVC) with ReadWriteMany (RWX) access mode. The specified 'BucketLoggingPVC' does not support RWX")
+		}
+
+		return util.NewPersistentError("InvalidBucketLoggingConfiguration",
+			"'Guaranteed' BucketLogging requires a Persistent Volume Claim (PVC) with ReadWriteMany (RWX) access mode. The specified 'BucketLoggingPVC' does not found")
+	}
+
+	util.KubeCheck(r.BucketLoggingPVC)
+	if err := r.prepareODFBucketLoggingPVC(); err != nil {
+		return err
+	}
+	r.Own(r.BucketLoggingPVC)
+
+	if r.BucketLoggingPVC.UID != "" {
```

Comment:
this check needs to come before prepareODFBucketLoggingPVC, if PVC already created we don't need to recreate it

Reply:
Done, Thanks.